crates/oxc_span/src/source_type.rs

```diff
@@ -166,25 +174,316 @@ impl SourceType {
+    }
+    fn consume(&mut self, target: &str) -> bool {
+        let mut chars = self.chars.clone();
+        for ch in target.chars() {
+            if let Some(c) = chars.next() {
+                if c != ch {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+
+        self.chars = chars;
+        true
+    }
+    fn consume_to(&mut self, target: char) -> bool {
+        for ch in self.chars.by_ref() {
+            if target == ch {
+                return true;
+            }
+        }
+
+        false
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::path::Path;
+
+    use crate::Span;
+
+    use super::SpecialLanguageVariantParser;
+
+    fn get_js_content<'a>(parser: &'a SpecialLanguageVariantParser) -> &'a str {
+        let start = parser.start;
+        let end = parser.end;
+        Span::new(start, end).source_text(parser.source_text)
+    }
+
+    #[test]
+    fn test_parse_vue_one_line() {
+        let source_text = r#"
+        <template>
+            <div>
+                <h1>hello world</h1>
+            </div>
+        </template>
+        <script> console.log("hi") </script>
+        "#;
+
+        let mut parser =
+            SpecialLanguageVariantParser::new(Path::new("test.vue"), source_text).unwrap();
+        parser.parse();
+        assert_eq!(get_js_content(&parser), r#" console.log("hi") "#);
+    }
+
+    #[test]
+    fn test_parse_vue_multi_line() {
+        let source_text = r#"
+        <template>
+            <div>
+                <h1>hello world</h1>
+            </div>
+        </template>
+        <script>
+            console.log("hi")
+            console.log("I am multi line")
+            console.log("<script></script>")
+            console.log(`<script></script>`)
+            console.log('<script></script>')
+        </script>
+        "#;
+
+        let mut parser =
+            SpecialLanguageVariantParser::new(Path::new("test.vue"), source_text).unwrap();
+        parser.parse();
+        assert!(!parser.is_ts);
+        assert_eq!(
+            get_js_content(&parser),
+            r#"
+            console.log("hi")
+            console.log("I am multi line")
+            console.log("<script></script>")
+            console.log(`<script></script>`)
+            console.log('<script></script>')
+        "#
+        );
+    }
+
+    #[ignore]
+    #[test]
+    fn test_parse_vue_with_ts_flag() {
+        let source_text = r#"
+        <script lang="ts" setup>
+            1/1 
+        </script>
+
+        <template>
+        </template>
+        "#;
+
+        let mut parser =
+            SpecialLanguageVariantParser::new(Path::new("test.vue"), source_text).unwrap();
+        parser.parse();
+        assert!(parser.is_ts);
+        assert_eq!(
+            get_js_content(&parser),
+            "
+                1/1
+            "
+        );
+    }
```

Comment:
TODO: need to pass this. Currently doesn't know whether `/` is the start of Reg Expression or an operator.

Reply:
Oh, we don't need to check '</script>' in  RegExpression, `/</script>/` is not a valid syntax..